### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by brandond**
Fixes multiple issues that came up while working with EIO to upgrade the K3s and RKE2 channel servers from v0.3.0 😬  to v0.7.0

* Set default prefix value
    There must be the same number of path prefixes as config keys. Unfortunately the default CLI args contain a single path prefix, but no subkey, which is invalid and results in startup failing with `level=fatal msg="keys-prefix lengths
are not equal [v1-release] [] 0.0.0.0:8080"`.
    Fix that by adding a default subkey of an empty string - so just the root of the document.
* Bump rancher/wrangler and rancher/apiserver
* Print subkey and prefix on startup
* Add support for Github token auth
    * Bump GH client version
    * Add support for GH token auth
    * Print status messages while loading initial config
    * Raise fatal error if initial config load fails, instead of silently serving with      no release data. This resolves an issue where the server would come up      but serve no data because the error from the initial load was silently
      ignored. The most frequent cause of this was rate limiting due to lack      of auth support:
      > `Failed to load config from [https://raw.githubusercontent.com/k3s-io/k3s/master/channel.yaml]: GET https://api.github.com/repos/k3s-io/k3s/releases: 403 API rate limit exceeded for CLIENT-IP. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,49 +8,51 @@ replace k8s.io/client-go => k8s.io/client-go v0.20.0
...
-	github.com/google/go-github/v29 v29.0.3
+	github.com/google/go-github/v67 v67.0.0
 	github.com/gorilla/handlers v1.4.2
-	github.com/gorilla/mux v1.8.0
+	github.com/gorilla/mux v1.8.1
 	github.com/pkg/errors v0.9.1
-	github.com/rancher/apiserver v0.0.0-20240124133512-e93bfa07e5f1
-	github.com/rancher/wrangler/v3 v3.0.0-rc2
+	github.com/rancher/apiserver v0.5.1
+	github.com/rancher/wrangler/v3 v3.0.1-rc.2
```
Comment (jiaqiluo) [inline go.mod]: Can you bump the version here to Wrangler's latest version v3.1.0, or v3.0.1 if that does not work? 

ref: https://github.com/rancher/wrangler/tags

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (brandond) [reply go.mod]: I set it to the same version as rancher/apiserver is using, but I guess v3.1 should be OK. 

I don't know why someone tagged it this way, but somehow the latest 3.0.1 RC tags are actually NEWER than 3.0.1:
image

Comment (brandond) [reply go.mod]: cc @tomleb can you comment on why we have rc tags that are newer than the patch release that they are an rc for?

Comment (brandond) [reply go.mod]: If we needed new RCs for 3.0.x, those should have been 3.0.2-rc.1 and 3.0.2-rc.2. From a semver perspective they are "lower" than 3.0.1 despite being newer.

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**